ClientApp/src/components/ListingsComp.js

```diff
@@ -41,33 +46,51 @@ export const ListingsComp = () => {
         }
     }
 
-  return (
-      <div style={{ overflow: 'hidden' }}>
-      {loading ? <p><em>Loading...</em></p>
-              : <><input
-                  type="text"
-                  placeholder="Search..."
-                  onChange={(event) => {
-                      setSearchTerm(event.target.value);
-                  } }
-              />
-                  <select onChange={(e) => sortArray(e.target.value)}>
-                      <option id="sortBy..">Sort by...</option>
-                      <option value="dateDesc">Date DESC</option>
-                      <option value="dateAsc">Date ASC</option>
-                  </select>
 
-                  <ul> {listingsData.filter((val) => {
-                      if (searchTerm == "") {
-                          return val
-                      }
-                      else if (val.name.toLowerCase().includes(searchTerm.toLowerCase()) ||
-                          val.location.country.toLowerCase().includes(searchTerm.toLowerCase()) ||
-                          val.location.city.toLowerCase().includes(searchTerm.toLowerCase())) {
-                          return val
-                      }
-                  }).map(data => <li key={data.id}><ListingComp {...data} refresh={fetchData} /></li>)} </ul></>
-      }
-    </div>
-  );
+    const indexOfLastListing = currentPage * listingsPerPage;
+    const indexOfFirstListing = indexOfLastListing - listingsPerPage;
+
+    const FilterListings = () => {
+        const filteredListings = listingsData.filter((val) => {
+            if (searchTerm == "") {
+                return val
+            }
+            else if (val.name.toLowerCase().includes(searchTerm.toLowerCase()) ||
+                val.location.country.toLowerCase().includes(searchTerm.toLowerCase()) ||
+                val.location.city.toLowerCase().includes(searchTerm.toLowerCase())) {
+                return val
+            }
+        });
+        setTotalListings(filteredListings.length);
```

Comment:
Browser throws a warning:
![image](https://user-images.githubusercontent.com/90030313/140741637-bf5910f9-ae06-4955-b9a4-9b255455de43.png)
I think it is due to the fact that when you render FilerListings it changes totalListings which causes the whole thing to rerender, which should rerender FilterListings again ... and there would be an infinite loop.
Although it doesn't crash, so I am probably wrong; I still think it is worth looking into the warning.

Reply:
Fixed!